apis/generate.go

```diff
@@ -83,7 +83,7 @@ limitations under the License.
 // (or protoc) to invoke them.
 
 //go:generate go install google.golang.org/protobuf/cmd/protoc-gen-go google.golang.org/grpc/cmd/protoc-gen-go-grpc
-//go:generate go run github.com/bufbuild/buf/cmd/buf@v1.31.0 generate
+//go:generate go run github.com/bufbuild/buf/cmd/buf@v1.53.0 generate
```

Comment:
I really don't like this `go run tool@version` pattern. Primarily because Earthly can't cache it, so it downloads the tool every time you (or CI) run `earthly +reviewable`, `earthly +generate`, etc.

I tried moving `buf` back into `go.mod` like our other dependencies, but sadly they still depend on half the known universe. We ended up having a build conflict in the Docker client (last time we tried we had a build conflict related to Otel).

Reply:
I've read that Go v1.24 may cache this style of `go run` (which I think would cause Earthly to cache them in turn). I'm reluctant to update to Go v1.24 though, because that'll force upgrading to golangci-lint v2.

Another thing we could try is use a separate module just for tools, and pass `-modfile ../tools/go.mod` to all of these commands. That way at least the tools could only conflict with each other, not Crossplane's 'real' dependencies.